#! /bin/sh
patch -p0 -f $* < $0
exit $?

2009-03-19 [colin]	3.7.1cvs22

	* src/toolbar.c
		Put spam button in default toolbar even if
		bogofilter and spamassassin are not built.
		Other anti-spam plugin could exist! :)
	* src/common/utils.c
		Fix exit status fetch on win32

Index: src/toolbar.c
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/toolbar.c,v
retrieving revision 1.43.2.107
retrieving revision 1.43.2.108
diff -u -p -u -r1.43.2.107 -r1.43.2.108
--- src/toolbar.c	23 Feb 2009 16:55:46 -0000	1.43.2.107
+++ src/toolbar.c	19 Mar 2009 16:35:36 -0000	1.43.2.108
@@ -489,10 +489,8 @@ static void toolbar_set_default_main(voi
 		{ A_SEPARATOR},
 		{ A_TRASH},
 #ifndef GENERIC_UMPC
-#if (defined(USE_SPAMASSASSIN_PLUGIN) || defined(USE_BOGOFILTER_PLUGIN))
 		{ A_LEARN_SPAM},
 #endif
-#endif
 		{ A_SEPARATOR},
 		{ A_GOTO_NEXT}
 	};
@@ -593,10 +591,8 @@ static void toolbar_set_default_msgview(
 		{ A_SEPARATOR},
 		{ A_TRASH},
 #ifndef GENERIC_UMPC
-#if (defined(USE_SPAMASSASSIN_PLUGIN) || defined(USE_BOGOFILTER_PLUGIN))
 		{ A_LEARN_SPAM},
 #endif
-#endif
 		{ A_GOTO_NEXT}
 	};
 	
Index: src/common/utils.c
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/common/utils.c,v
retrieving revision 1.36.2.168
retrieving revision 1.36.2.169
diff -u -p -u -r1.36.2.168 -r1.36.2.169
--- src/common/utils.c	17 Mar 2009 18:04:04 -0000	1.36.2.168
+++ src/common/utils.c	19 Mar 2009 16:35:36 -0000	1.36.2.169
@@ -3286,18 +3286,25 @@ static gint execute_sync(gchar *const ar
 
 	cm_return_val_if_fail(argv != NULL && argv[0] != NULL, -1);
 
+#ifdef G_OS_UNIX
 	if (g_spawn_sync(NULL, (gchar **)argv, NULL, G_SPAWN_SEARCH_PATH,
 			 NULL, NULL, NULL, NULL, &status, NULL) == FALSE) {
 		g_warning("Couldn't execute command: %s\n", argv[0]);
 		return -1;
 	}
 
-#ifdef G_OS_UNIX
 	if (WIFEXITED(status))
 		return WEXITSTATUS(status);
 	else
 		return -1;
 #else
+	if (g_spawn_sync(NULL, (gchar **)argv, NULL, G_SPAWN_SEARCH_PATH| 
+			 G_SPAWN_CHILD_INHERITS_STDIN|G_SPAWN_LEAVE_DESCRIPTORS_OPEN,
+			 NULL, NULL, NULL, NULL, &status, NULL) == FALSE) {
+		g_warning("Couldn't execute command: %s\n", argv[0]);
+		return -1;
+	}
+
 	return status;
 #endif
 }
